File: BST/python3/node.py
class Node:
    def __init__(self, value) -> None:
        self.value = value
        self.left: "Node" | None = None
        self.right: "Node" | None = None

    def __eq__(self, other):
        return self.value == other.value

    def __ne__(self, other):
        return self.value != other.value

    def __lt__(self, other):
        return self.value < other.value

    def __le__(self, other):
        return self.value <= other.value

    def __gt__(self, other):
        return self.value > other.value

    def __ge__(self, other):
        return self.value >= other.value

    def add(self, value) -> bool:
        if value < self.value:
            if self.left is None:
                self.left = Node(value)
                return True
            return self.left.add(value)

        if value > self.value:
            if self.right is None:
                self.right = Node(value)
                return True
            return self.right.add(value)

        return False

    def contain(self, value) -> bool:
        if value < self.value:
            if self.left is None:
                return False
            return self.left.contain(value)

        if value > self.value:
            if self.right is None:
                return False
            return self.right.contain(value)

        return True

    def length(self) -> int:
        a = 0 if self.left is None else self.left.length()
        b = 0 if self.right is None else self.right.length()
        return a + b + 1

    def height(self) -> int:
        return max(
            self.left.height() + 1 if self.left is not None else 1,
            self.right.height() + 1 if self.right is not None else 1,
        )

File: BST/c#/Node.cs
class Node
{
    public int value;
    private Node left;
    private Node right;

    public Node(int value){
        this.value = value;
    }

    public bool add(int value){
        if(value < this.value){
            if(this.left == null){
                this.left = new Node(value);
                return true;
            }
            return this.left.add(value);
        }

        if(value > this.value){
            if(this.right == null){
                this.right = new Node(value);
                return true;
            }
            return this.right.add(value);
        }

        return false;
    }

    public bool contain(int value){
        if (value < this.value){
            return this.left == null ? false : this.left.contain(value);
        }

        if (value > this.value){
            return this.right == null ? false : this.right.contain(value);
        }

        return true;
    }

    public int length(){
        int a = this.left == null ? 0 : this.left.length();
        int b = this.right == null ? 0 : this.right.length();
        return a+b+1;
    }

    public int height(){
        int a = this.left == null ? 1 : this.left.height() + 1;
        int b = this.right == null ? 1 : this.right.height() + 1;
        return a > b ? a : b;
    }
}
